lib/jekyll-admin/server/collection.rb

```diff
@@ -47,26 +49,70 @@ def collection
         collection[1] if collection
       end
 
+      def request_path
+        sanitized_path File.join(site.source, request_payload["path"])
+      end
+
+      def filename
+        "#{params["path"]}.#{params["ext"]}"
+      end
+
       def document_id
-        params["splat"].first.gsub(%r!(\d{4})/(\d{2})/(\d{2})/(.*)!, '\1-\2-\3-\4')
+        path = "#{params["splat"].first}/#{filename}"
+        path.gsub(%r!(\d{4})/(\d{2})/(\d{2})/(.*)!, '\1-\2-\3-\4')
       end
 
       def document_path
         sanitized_path File.join(collection.relative_directory, document_id)
       end
 
+      def relative_document_path
+        if params["splat"].first.empty?
+          File.join(collection.relative_directory, filename)
+        else
+          relative_to_collection = File.join(params["splat"].first, filename)
+          File.join(collection.relative_directory, relative_to_collection)
+        end
+      end
```

Comment:
Let's say `request_payload["path"]` for `/:collection_id/*?/?:path.:ext` endpoint is `_posts/some_folder/2017-01-01-test.md`. We need to compare this path with the one generated from `params`fields to check if there exists a rename. Any easy solution to achieve this?